presto-hive/src/main/java/com/facebook/presto/hive/HiveConnectorFactory.java

```diff
@@ -120,12 +123,20 @@ public Connector create(String connectorId, Map<String, String> config)
                             new HiveMetastoreAuthenticationKerberos.Module()),
                     installModuleIf(
                             HiveClientConfig.class,
-                            HiveClientConfig::isHdfsSaslEnabled,
-                            new HdfsAuthenticatingConnectorModule()),
+                            c -> c.getHdfsAuthenticationType() == AuthenticationType.SIMPLE,
+                            new SimpleConnectorModule()),
                     installModuleIf(
                             HiveClientConfig.class,
-                            hiveClientConfig -> !hiveClientConfig.isHdfsSaslEnabled(),
-                            new SimpleConnectorModule()),
+                            c -> c.getHdfsAuthenticationType() == AuthenticationType.SIMPLE_IMPERSONIFICATION,
+                            new SimpleImpersonificationConnectorModule()),
+                    installModuleIf(
+                            HiveClientConfig.class,
+                            c -> c.getHdfsAuthenticationType() == AuthenticationType.KERBEROS,
```

Comment:
Maybe use static imports in here? I think `c.getHdfsAuthenticationType() == SIMPLE_IMPERSONIFICATION` is still pretty readable and less verbose.